worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/DisallowedBlocksExtent.java

```diff
@@ -134,34 +133,68 @@ public IChunkSet processSet(final IChunk chunk, final IChunkGet get, final IChun
             if (!set.hasSection(layer)) {
                 continue;
             }
-            char[] blocks = Objects.requireNonNull(set.loadIfPresent(layer));
-            it:
-            for (int i = 0; i < blocks.length; i++) {
-                char block = blocks[i];
-                BlockState state = states[block];
-                if (blockedBlocks != null) {
-                    if (blockedBlocks.contains(state.getBlockType().getId())) {
-                        blocks[i] = 0;
+
+            if(!(set instanceof IntSetBlocks)) {
+                char[] blocks = Objects.requireNonNull(set.loadCharsIfPresent(layer));
+                it:
+                for (int i = 0; i < blocks.length; i++) {
+                    char block = blocks[i];
+                    BlockState state = states[block];
```

Comment:
Is there a particular reason you reuse the same code in the `else` block, instead of creating a private method?

Reply:
I attempted to convert the if/else block into a private method, but I had difficulty finding an implementation that worked due to the use of a labeled loop and the char/int difference in types the two blocks used. I did find an error in the bottom of the int block with ```state.getOrdinalChar()``` in that it should be ```state.getOrdinal()``` which I will be adding with my next commit.